barterdude/conf.py

```diff
@@ -3,10 +3,13 @@
 from pythonjsonlogger import jsonlogger
 
 
-handler = logging.StreamHandler()
-handler.setFormatter(jsonlogger.JsonFormatter(
-    '(levelname) (name) (pathname) (lineno)',
-    timestamp=True
-))
-logger = logging.getLogger('barterdude')
-logger.addHandler(handler)
+def getLogger(name="barterdude", level=logging.INFO):
+    handler = logging.StreamHandler()
+    handler.setFormatter(jsonlogger.JsonFormatter(
+        '(levelname) (name) (pathname) (lineno)',
+        timestamp=True
+    ))
+    logger = logging.getLogger(name)
+    logger.setLevel(level)
+    logger.addHandler(handler)
```

Comment:
Se essa função for chamada mais de uma vez para o mesmo nome vai acabar adicionando vários handlers no mesmo logger? A gente não deveria evitar essa situação?

Reply:
good point! Fixed! \o/

Reply:
A função continua adicionando o mesmo handler várias vezes no mesmo logger.

Reply:
Ele mantém somente um handler quando é igual: https://github.com/olxbr/BarterDude/pull/21/files#diff-d4caf9d941d0df1a89ca865431c1e468R55